### DISCUSSION TITLE
Retrieving Validation Results and input data columns at an individual record level

### DISCUSSION THREAD

**Discussion by n-janani** [great-expectations/great_expectations#5070] [Support] [answered]
Hello team! I am very new to GE, and am currently evaluating it for automated data quality in our data pipelines.
One of our requirements is to be able to track validation results at an individual record level. 
To elaborate: If I passed in a data frame of 10 records of the source data, to GE, I want to be able to capture the validation results (success status, expectation applied etc) for each of those 10 records, and also have the input columns added to these validation results.
For example: if my data frame had columns "row_id, column_name", and I wanted GE to run a check on "column_name", then after the validations are run, I want the output to have all the 10 rows and look like "row_id, column_name, expectation_name, success (true/false status)". Is this something I can configure GE to do? 
Thank you in advance for your time and help!

**Comment by austiezr** [comment, marked as answer]
Hey @n-janani ! Thanks for reaching out. Great Expectations doesn't currently support the creation of a results object as described above. However, `result_format` may approximate the functionality you're looking. See [this document](https://docs.greatexpectations.io/docs/reference/expectations/result_format) for reference on how to configure `result_format`, and what to expect from it.
